### PR TITLE
Flushed out the Java mobile ecosystem

### PR DESCRIPTION
**Pull request by codenameone**
Added Avian, Codename One/ParparVM & J2ObjC to the list as they were all missing

### CODE REVIEW

**Comment by akullpp** [comment]
Please remove the "Open Source" parts in your descriptions since it is implied. Only commercial products have to be mentioned explicitly.

File: README.md [hunk 1]
```diff
@@ -442,12 +442,16 @@ A curated list of awesome Java frameworks, libraries and software.
 
 *Everything else.*
 
+* [Avian](https://github.com/ReadyTalk/avian) - Open source JVM with both a JIT & AOT modes. Includes an iOS port.
+* [Codename One](https://www.codenameone.com/) - Open source cross platform solution for writing native mobile apps in Java for iOS, Android, Windows etc.
```
Comment (akullpp) [inline README.md]: Please be as concise as possible, i.e.

`Cross platform solution for writing native mobile apps.`

File: README.md [hunk 2]
```diff
@@ -442,12 +442,16 @@ A curated list of awesome Java frameworks, libraries and software.
 
 *Everything else.*
 
+* [Avian](https://github.com/ReadyTalk/avian) - Open source JVM with both a JIT & AOT modes. Includes an iOS port.
```
Comment (akullpp) [inline README.md]: Remove the "Open source", see comment in Conversations. Also we have a JVM category, move the entry.

File: README.md [hunk 3]
```diff
@@ -442,12 +442,16 @@ A curated list of awesome Java frameworks, libraries and software.
...
+* [Codename One](https://www.codenameone.com/) - Open source cross platform solution for writing native mobile apps in Java for iOS, Android, Windows etc.
 * [Design Patterns](https://github.com/iluwatar/java-design-patterns) - Implementation and explanation of the most common design patterns.
+* [J2ObjC](https://github.com/google/j2objc) - Java to Objective-C translator for porting Android libraries to iOS.
 * [Jimfs](https://github.com/google/jimfs) - In-memory file system.
 * [Lanterna](https://code.google.com/p/lanterna/) - Easy console text GUI library similar to curses.
 * [LightAdmin](http://lightadmin.org/) - Pluggable CRUD UI library for rapid application development.
 * [Modern Java - A Guide to Java 8](https://github.com/winterbe/java8-tutorial) - Popular Java 8 guide.
 * [OpenRefine](http://openrefine.org/) - Tool for working with messy data: cleaning, transforming, extending it with web services and linking it to databases.
+* [ParparVM](https://github.com/codenameone/CodenameOne/tree/master/vm) - The open source VM that underlies [Codename One](https://www.codenameone.com/).
```
Comment (akullpp) [inline README.md]: Move to JVM section and mention the unique advantage of the JVM which is non-blocking concurrent GC for iOS.

**Comment by codenameone** [comment]
Did you notice that I made the changes requested? Any further issues?
Thanks!

**Comment by akullpp** [comment]
Thanks great work.

**Pull request merged**